packages/drivers/odsp-driver/src/graph.ts

```diff
@@ -243,35 +280,44 @@ export async function getGraphItemLite(
             const partialUrl = `${slashTerminatedOriginOrEmptyString(msGraphOrigin)
                 }drives/${driveId}/items/${itemId}?select=webUrl,webDavUrl,name`;
 
-            let response: Response | undefined;
+            let response: IOdspResponse<GraphItemLite> | undefined;
             try {
                 response = await graphFetch(getShareLinkToken, siteUrl, partialUrl, "GetGraphItemLite", logger);
-            } catch { }
-
-            // Cache only if we got a response and the response was a 200 (success) or 404 (NotFound)
-            if (!response || (response.status !== 200 && response.status !== 404)) {
-                graphItemLiteCache.remove(cacheKey);
-                return undefined;
-            } else {
-                try {
-                    const result: any = await response.json();
-                    if (result && result.webDavUrl && result.name) {
-                        const liteGraphItemInfo: GraphItemLite = {
-                            webUrl: result.webUrl,
-                            webDavUrl: result.webDavUrl,
-                            name: result.name,
-                        };
-                        return liteGraphItemInfo;
-                    } else {
-                        return undefined;
-                    }
-                } catch (error) {
+            } catch(error) {
+                // Cache only if we got a response and the response was a 200 (success) or 404 (NotFound)
+                if (error.statusCode !== 404) {
```

Comment:
The comment and code do not match. Why single out 404? If user has no permissions (401), it should be handled same as 404 (which could mean no permissions, btw, not just item does not exist).

Reply:
Changed it to remove from cache and treat 404 like others.